### DISCUSSION TITLE
how to push the local change to another new branch

### DISCUSSION THREAD

**Discussion by jasontsaitks** [actions/checkout#328] [General]
my goal is:
In repoA,
1. checkout from repoB
2. do some work, generate some files
3. commit change
4. push to a new branch in repoB and create pull request in repoB

can this be done?

**Comment by alsuren** [comment]
Here is some code to do everything apart from the pull-request bit: https://github.com/MHRA/products/blob/master/.github/workflows/medicines-api-master.yaml#L85-L118

https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/security-hardening-for-github-actions#considering-cross-repository-access suggests that you should use generate a "Repository deploy key" from repoB, and store it in the secrets of repoA. (this will then be used either as `token` as MHRA does it, or `ssh-key`. I can't remember off the top of my head).

Once you have pushed your commits, you will have to use the github api to make the pull request. I'll let you work that one out on your own. :D 

... I just noticed that this was posted 121 days ago. I guess you solved it already?
